app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,46 @@
+# Comments model belongs to book and to user.
+class CommentsController < ApplicationController
+  def show
+    comment = Comment.find(params[:id])
+    render json: comment, status: :ok
+  end
+
+  def index
+    comments = Comment.where(book_id: params[:book_id])
+    render json: comments, status: :ok
+  end
+
+  def create
+    authorize :comment
+    book = Book.find(params[:book_id])
```

Comment:
put this inside a function like this
`def book`
`    @book ||= Book.find(params[:book_id])`
`end`